### ISSUE TITLE
Rename DeploymentFramework to DocumentFramework

### ISSUE THREAD

**Issue by ml019** [hamlet-io/engine#1472]
## Expected Behavior
The use of deploymentFrameworks in situations like diagrams, which do not involve deployments, highlight that this concept is not correctly named.

More correctly, a deployment framework currently represents a particular type of document and support functions and macros to make their creation easier. Examples are cloud formation templates, arm templates and diagrams.

It is proposed to rename deploymentFramework to documentFramework.

While largely an internal change, the rename will make it easier to document the purpose of this idea within the Hamlet framework.

## Current Behavior
The term "deploymentFramework" is used to describe facilities to create one or more types of document. The use of the name has gone beyond its original design. 

## Possible Solution
Rename to documentFramework

**Comment by RossMurr4y** [comment]
Does `DocumentFramework` lean too far the other way (towards "this produces documentation")?

The 3 `DeploymentFramework` directories across shared/aws/azure providers currently only contain an `output.ftl` file. Output data structures are what is being defined there - does `Outputs` capture the same idea?

**Comment by ml019** [comment]
I discussed this with @roleyfoley and we came to the same conclusion, namely that we could drop the concept of a deployment framework as such and instead just have output types. 

Removal has some implications for the way state is managed. The proposal was to 
- incorporate the output type into the file naming
- remove the deploymentframework as a level in the state hierarchy 

We partly have this already with the naming of files for subsets so it seems a natural next step.

I'll close this issue and create a new one focussed on reworking the handling of output types.
